### ISSUE TITLE
intermittent IMAP failure

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7601]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)| @warner reported|
|-|-|
|Trac ID|trac#369|
|Type|defect|
|Created|2003-10-27 03:09:01Z|

<details><summary>Searchable metadata</summary>

```
trac-id__369 369
type__defect defect
reporter__warner warner
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1067224141000000 1067224141000000
changetime__1469103451417242 1469103451417242
version__ 
owner__ 
cc__exarkun cc__warner
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="warner's avatar" src="https://avatars.githubusercontent.com/u/27146?s=50" width="50" height="50">](https://github.com/warner)|@warner commented|
|-|-|

```
#!html
<pre>
This one happened once, on the freebsd buildslave. I haven't seen it occur
anywhere else.
http://www.twistedmatrix.com/buildbot/freebsd/events/1027/summary

FAILURE: testFetchFlagsUID (twisted.test.test_imap.NewFetchTestCase)
-------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/slave/twisted/bsd-full2.2/Twisted/twisted/trial/unittest.py", line
252, in _runPhase
    stage(*args, **kwargs)
  File "/home/slave/twisted/bsd-full2.2/Twisted/twisted/trial/util.py", line 43,
in reactorCleanUp
    raise unittest.FailTest, msg
FailTest: 
pendingTimedCalls still pending:
 &lt;DelayedCall 144995308 [59.884032011s] called=0 cancelled=0
IMAP4Server.__timedOut()>

===============================================================================
ERROR: testFetchFull (twisted.test.test_imap.NewFetchTestCase)
-------------------------------------------------------------------------------
Traceback: twisted.internet.error.AlreadyCancelled, Tried to cancel an
already-cancelled event.
/home/slave/twisted/bsd-full2.2/Twisted/twisted/python/log.py:65:callWithLogger
/home/slave/twisted/bsd-full2.2/Twisted/twisted/python/log.py:52:callWithContext
/home/slave/twisted/bsd-full2.2/Twisted/twisted/python/context.py:53:callWithContext
/home/slave/twisted/bsd-full2.2/Twisted/twisted/python/context.py:32:callWithContext
/home/slave/twisted/bsd-full2.2/Twisted/twisted/internet/default.py:524:_doReadOrWrite
/home/slave/twisted/bsd-full2.2/Twisted/twisted/internet/tcp.py:289:connectionLost
/home/slave/twisted/bsd-full2.2/Twisted/twisted/protocols/imap4.py:518:connectionLost
/home/slave/twisted/bsd-full2.2/Twisted/twisted/protocols/policies.py:416:setTimeout
/home/slave/twisted/bsd-full2.2/Twisted/twisted/internet/base.py:79:cancel
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

```
#!html
<pre>
Still a valid issue?
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

```
#!html
<pre>
I ran the whole test_imap.py module 218 times with no failures.  Of course, I
didn't do it on a FreeBSD box.

I'm rather tempted to resolve this issue.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

```
#!html
<pre>
This is really "WONTFIX", until somebody who cares about FreeBSD comes along
again.  Without a buildbot, it's impossible to tell why this would fail
(especially considering it doesn't fail on the BSD-derived OS X)
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub commented|
|-|-|

In changeset 127aa16b0a60b7e1b505017ec89c637b4e603fbb
```
#!CommitTicketReference repository="" revision="127aa16b0a60b7e1b505017ec89c637b4e603fbb"
Merge pull request #369 from twisted/8635-rodrigc-udp-docstring

Author: rodrigc
Reviewer: scop
Fixes: #8635

Clarify docstring for twisted.internet.udp.Port.write and twisted.internet.udp.Port.writeSequence
```
